pennylane/devices/default_qubit.py

```diff
@@ -297,7 +299,7 @@ def _apply_z(self, state, axes, **kwargs):
         Returns:
             array[complex]: output state
         """
-        return self._apply_phase(state, axes, -1)
+        return self._apply_phase(state, axes, -1.0)
```

Comment:
How come an integer worked before? :thinking:

Reply:
It was a fixture I used before adding `_const_mul`. I don't remember correctly, but it somehow converted the data type of an array. As `_apply_phase` now uses `_const_mul`, which doesn't convert array type, I will revert this change.

Reply:
It was a fixture I used before adding `_const_mul`. I remember that it somehow converted the data type of an array. As `_apply_phase` now uses `_const_mul`, which doesn't convert array type, I think I can safely revert this change.